app/scripts/metamask-controller.js

```diff
@@ -2618,9 +2671,12 @@ export default class MetamaskController extends EventEmitter {
    * @param {string} origin - The origin identifier for which network state is requested (default: 'metamask').
    * @returns {object} An object containing important network state properties, including chainId and networkVersion.
    */
-  getProviderNetworkState(origin = 'metamask') {
+  getProviderNetworkState(origin = METAMASK_DOMAIN) {
     let chainId;
-    if (this.preferencesController.getUseRequestQueue()) {
+    if (
```

Comment:
Just noticed a bug in this function: the `networkVersion` returned is always for the globally selected network, even when this feature is on. That would result in the provider having a `networkId` that contradicts the `chainId`. Not good.

Not a blocker for this PR since it wasn't introduced here, but we should fix this as well.

Reply:
Heres a ticket: https://github.com/MetaMask/MetaMask-planning/issues/2176